pkg/ddl/notifier/events.go

```diff
@@ -426,15 +427,77 @@ func NewFlashbackClusterEvent() *SchemaChangeEvent {
 	}
 }
 
+// NewDropSchemaEvent creates a schema change event whose type is ActionDropSchema.
+func NewDropSchemaEvent(dbInfo *model.DBInfo, tables []*model.TableInfo) *SchemaChangeEvent {
+	miniTables := make([]*MiniTableInfoForSchemaEvent, len(tables))
+	for i, table := range tables {
+		miniTables[i] = &MiniTableInfoForSchemaEvent{
+			ID:   table.ID,
+			Name: table.Name,
+		}
+		if table.Partition != nil {
+			for _, part := range table.Partition.Definitions {
+				miniTables[i].Partitions = append(miniTables[i].Partitions, &MiniPartitionInfoForSchemaEvent{
```

Comment:
you have prepare the slice capacity in line 432 but not here 🤔 I think a table may have many partitions so it's also good to do it.

Reply:
Updated.